Lib/xml/dom/minidom.py

```diff
@@ -300,12 +300,27 @@ def _in_document(node):
         node = node.parentNode
     return False
 
-def _write_data(writer, data):
+def _write_data(writer, text, attr):
```

Comment:
Is there a reason why this can't be two functions? One for regular text content, one for attributes? Adding a flag just, well, adds a flag for extra complexity, instead of just saying `_write_attribute_data()` or `_write_text_data()`. Especially since the value that you pass into the flag argument is just a bare `True` or `False`, without any indication what is set to true or false here.

Reply:
No other reasons except following the style of the old code and the fact that most of the code is shared between functions. I would perhaps merge these function in ElentText.py as well if I wrote them from scratch.

It is an internal function called just in two places, so I did not even bother to pass the argument by keyword.

Reply:
> pass the argument by keyword

I also wouldn't have used keyword arguments here, mostly for performance reasons. I'll leave the final decision to you. Personally, I would avoid passing bare constants like `None` or `True` as call arguments in newly written code (and change them when touching old code), because they reduce the readability for me. It's just not clear what they mean in the given call without looking up the function signature, so they interrupt my code reading process.

Reply:
I completely agree with you. I prefer to make boolean parameters keyword-only.

An exception was made here for performance reasons, and it was only possible because it is an internal function called in two places.